tdmpc2/common/math.py

```diff
@@ -72,9 +72,9 @@ def two_hot(x, cfg):
 	x = torch.clamp(symlog(x), cfg.vmin, cfg.vmax).squeeze(1)
 	bin_idx = torch.floor((x - cfg.vmin) / cfg.bin_size).long()
 	bin_offset = ((x - cfg.vmin) / cfg.bin_size - bin_idx.float()).unsqueeze(-1)
-	soft_two_hot = torch.zeros(x.size(0), cfg.num_bins, device=x.device)
-	soft_two_hot.scatter_(1, bin_idx.unsqueeze(1), 1 - bin_offset)
-	soft_two_hot.scatter_(1, (bin_idx.unsqueeze(1) + 1) % cfg.num_bins, bin_offset)
+	soft_two_hot = x.new_zeros(x.shape[0], cfg.num_bins)
+	soft_two_hot = soft_two_hot.scatter(1, bin_idx.unsqueeze(1), 1 - bin_offset)
+	soft_two_hot = soft_two_hot.scatter(1, (bin_idx.unsqueeze(1) + 1) % cfg.num_bins, bin_offset)
```

Comment:
same: avoid in-place ops. `new_zeros` isn't very different than the `zeros` before, just avoid reading the device of x directly.